### PR TITLE
Code documentation

### PR DESCRIPTION
**Pull request by aditya0by0**
- PR for Issue #9 

# Tasks
- Annotate code with docstrings for relevant classes/code in the repository
- Add Type Hints for relevant classes/code in the repository

### CODE REVIEW

**Comment by aditya0by0** [comment]
- #### Due to dependency with PR #29,  Issue #10 the below classes/code are yet to be updated for **Docstrings** and **Type Hints**
>- `datasets/chebi.py`
>- 2 Test Classes related to chebi data

Also, the PR #29 has been re-format with new pre-commit, so I think this PR should be merged after the PR #29, with potential conflicts reolved

File: chebai/cli.py [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+    Args:
+        save_config_kwargs (dict): Keyword arguments for saving configuration.
+        parser_kwargs (dict): Keyword arguments for parser configuration.
+
+    Attributes:
+        save_config_kwargs (dict): Configuration options for saving.
+        parser_kwargs (dict): Configuration options for the argument parser.
+    """
+
+    def __init__(self, save_config_kwargs: dict, parser_kwargs: dict):
```
Comment (sfluegel05) [inline chebai/cli.py:19]: It is not clear to me why you changed the parameters here. With `**kwargs`, you can do the same (and also use other parameters for the LightningCLI class).

Comment (aditya0by0) [reply chebai/cli.py]: Sorry for the confusion, it was done by mistake. You are correct that **kwargs can be used to handle the same parameters along with others for the LightningCLI class. The original parameters should not have been modified. Thank you for pointing it out.

I have restored the original parameters for it now.

**Comment by aditya0by0** [comment]
> * #### Due to dependency with PR [Data handling restructure #29](https://github.com/ChEB-AI/python-chebai/pull/29),  Issue [Data handling needs to be restructured #10](https://github.com/ChEB-AI/python-chebai/issues/10) the below classes/code are yet to be updated for **Docstrings** and **Type Hints**
> 
> > * `datasets/chebi.py`
> > * 2 Test Classes related to chebi data
> 
> Also, the PR #29 has been re-formated with new pre-commit, so I think this PR should be merged after the PR #29, with potential conflicts reolved

Hi @sfluegel05, I have resolved the conflicts and updated the mentioned files with docstrings and type hints. Please review.

**Comment by sfluegel05** [comment]
Thanks, that is a really comprehensive documentation. We will profit from that for a long time.

**Pull request merged**